src/context/StoreContext.js

```diff
@@ -56,23 +58,42 @@ export default function StoreContextProvider(props) {
     null
   )
 
+  const onBibleRefUIChange = (bId, ch, v) => {
+    console.log(`\n### Reference changed to ${bId} - ${ch}:${v}\n\n`);
+    const bRefId = "BibleRef"
+    if (cacheBRef 
+      && (cacheBRef?.bookId?.toUpperCase() === bId?.toUpperCase())
+      && (cacheBRef.chapter === ch) 
+      && (cacheBRef.verse === v)) { // suppress cached external updates
+        cacheBRef = {}
+    } else if ((bibleReference?.bookId !== bId?.toUpperCase()) 
+      || (bibleReference.chapter !== ch) 
+      || (bibleReference.verse !== v)) { // Do not re-trigger new events
+        localSetBibleReference({ sourceId: bRefId, bookId: bId.toUpperCase(), chapter: ch, verse: v })
+    }
+  }
+
+  const { state: bRefState, actions: bRefActions } = useBibleReference({
+    initialBook: 'mat',
```

Comment:
I would recommend moving the hard coded initial reference to `@common/constants` this way we can easily modify the logic behind the initial reference if we need to in the future

Reply:
Done